entc/gen/type.go

```diff
@@ -336,11 +336,7 @@ func (t Type) PackageAlias() string { return t.alias }
 // Receiver returns the receiver name of this node. It makes sure the
 // receiver names doesn't conflict with import names.
 func (t Type) Receiver() string {
-	r := receiver(t.Name)
-	if t.Package() == r {
-		return "_" + r
-	}
-	return r
+	return "m"
```

Comment:
This should be the model prefix imo. But what will happen for all those who use custom templates? The previous receiver should still work, so I believe we should detect this there as well.

Reply:
@a8m There problem is, the issue https://github.com/ent/ent/issues/4351 has conflict on this model receiver, not the create/update/delete/query one. So it must be change to static to resolve https://github.com/ent/ent/issues/4351

We can add option to allow the user fallback to old behavior when upgrading. But, honestly, I prefer them to update the custom template, if the methods on Ent's model has variable that conflicting with `m`.

Reply:
> @a8m There problem is, the issue https://github.com/ent/ent/issues/4351 has conflict on this model receiver, not the create/update/delete/query one. So it must be change to static to resolve https://github.com/ent/ent/issues/4351

But we can fix the template that is used by #4351. We have such checks in the template like there: https://github.com/ent/ent/blob/master/entc/gen/template/client.tmpl#L287 (e.g., we can replace ok with _ok). 

I just try to avoid breaking changes to user templates.

Reply:
@a8m, I actually wanna remove all check like you mention. If we just move to static one, we don't need these check anymore, for all places.

> I just try to avoid breaking changes to user templates.

I prefer to mention this in the release and ask the user update their template if they using custom one.

Reply:
Add check for #4351 is an Ad-hoc, and we will need to update for other if there is another conflict. Which still a pan for us.

Reply:
Other option, that use `_m` as receiver. It will avoid the conflict with existing custom template.

Reply:
You can also change `receiver` to prefix with `_`.